keps/sig-cli/5295-kyaml/README.md

```diff
@@ -0,0 +1,1230 @@
+
+NOTHING
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A
+
+### Rollout, Upgrade and Rollback Planning
+
+N/A
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+N/A
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+NO
+
+### Monitoring Requirements
+
+N/A
+
+###### How can an operator determine if the feature is in use by workloads?
+
+N/A
+
+###### How can someone using this feature know that it is working for their instance?
+
+N/A
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+NO
+
+### Dependencies
+
+NONE
+
+###### Does this feature depend on any specific services running in the cluster?
+
+NONE
+
+### Scalability
+
+N/A
+
+###### Will enabling / using this feature result in any new API calls?
+
+NO
+
+###### Will enabling / using this feature result in introducing new API types?
+
+NO
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+NO
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+NO
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+NO
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+Using this may incur some client-side resource usage, but marginal.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+NO
+
+### Troubleshooting
+
+If someone thinks the KYAML rendering is wrong they can instead render to plain YAML or JSON and compare.  Additionally, they can render to KYAML and then use yamlfmt to convert that to plain YAML.
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+N/A
+
+###### What are other known failure modes?
+
+It’s possible we missed some esoteric structure that renders incorrectly.  Those will be handled as bugs.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+May 09, 2025: KEP draft v0
+
+## Drawbacks
+
+Having a local dialect of YAML could be confusing to people who are used to conventional YAML.
+
+Trying to text-patch KYAML with plain YAML (as in a help chart) will almost certainly not work.
```

Comment:
```suggestion
Trying to text-patch KYAML with plain YAML (as in a helm chart) will almost certainly not work.
```

Will not work in which sense? It wouldn't be KYAML anymore, but should still be valid YAML?

Reply:
If you were writing flow-style YAML and you try to text-substitute it into YAML which has switched to block style it is likely invalid. So if your helm chart decided to switch then you would have to adapt.

That said, text-templating YAML instead of using patches or some other structure aware approach is _always_ fragile.

Reply:
Added a note